app/org/sagebionetworks/bridge/models/schedules/ActivityScheduler.java

```diff
@@ -39,39 +44,32 @@ protected DateTime getScheduledTimeBasedOnEvent(ScheduleContext context) {
         return eventTime;
     }
     
-    protected void addScheduledActivityForAllTimes(List<ScheduledActivity> scheduledActivities, 
-            SchedulePlan plan, ScheduleContext context, DateTime scheduledTime) {
-        if (schedule.getTimes().isEmpty()) {
-            // We're using whatever hour/minute/seconds were in the original event.
-            addScheduledActivityAtTime(scheduledActivities, plan, context, scheduledTime);
-        } else {
-            for (LocalTime time : schedule.getTimes()) {
-                scheduledTime = new DateTime(scheduledTime, context.getZone()).withTime(time);
-                addScheduledActivityAtTime(scheduledActivities, plan, context, scheduledTime);
-            }
+    protected void addScheduledActivityForAllTimes(List<ScheduledActivity> scheduledActivities, SchedulePlan plan,
+            ScheduleContext context, LocalDate localDate) {
+        
+        List<LocalTime> localTimes = (schedule.getTimes().isEmpty()) ? MIDNIGHT_IN_LIST : schedule.getTimes();
+        for (LocalTime localTime : localTimes) {
+            addScheduledActivityAtTime(scheduledActivities, plan, context, localDate, localTime);
         }
     }
     
     protected void addScheduledActivityAtTime(List<ScheduledActivity> scheduledActivities, SchedulePlan plan,
-            ScheduleContext context, DateTime scheduledTime) {
-        // Assert that the scheduledTime was constructed by subclass implementation with the correct time zone.
-        checkArgument(context.getZone().equals(scheduledTime.getZone()), 
-            "Scheduled DateTime does not have requested time zone: " + scheduledTime.getZone());
-
-        // If this time point is outside of the schedule's active window, skip it.
-        if (isInWindow(scheduledTime)) {
+            ScheduleContext context, LocalDate localDate, LocalTime localTime) {
+        
+        if (isInWindow(localDate, localTime)) {
             // As long at the activities are not already expired, add them.
-            DateTime expiresOn = getExpiresOn(scheduledTime);
+            DateTime expiresOn = getExpiresOn(localDate, localTime);
```

Comment:
This is a good observation and as a refactor, fairly straightforward to do, and the results are better. But it's going to add a lot of changes on top of what's already here. I'm inclined to file a JIRA and deal with it after this is rolled out.